drivers/clock_control/clock_control_mcux_ccm.c

```diff
@@ -137,6 +161,13 @@ static int mcux_ccm_on(const struct device *dev,
 		CLOCK_EnableClock(sai_clocks[instance]);
 		return 0;
 #endif
+#if defined(CONFIG_SOC_MIMX8ML8_ADSP)
+	case IMX_CCM_SAI1_CLK:
```

Comment:
You can group this code together with `#if defined(CONFIG_SOC_MIMX8QM6_ADSP) || defined(CONFIG_SOC_MIMX8QX6_ADSP)` above.

Reply:
I think this would break as soon as you have ARRAY_SIZE(sai_clocks) (for 8QM/8QXP) != ARRAY_SIZE(sai_clocks) (for 8MP) and user misconfigures the DTS. If that's alright then I can make the change.

Reply:
Ok, I understand. works for me.